packages/@react-spectrum/dialog/docs/DialogTrigger.mdx

```diff
@@ -0,0 +1,464 @@
+        This is a popover offset by an additional 100px to the right of the trigger.
+      </Text>   
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+### Flipping
+
+Applying `shouldFlip` to the DialogTrigger makes the Popover Dialog attempt to flip
+on its main axis in situations where the original placement would cause it to render out of view.
+
+```tsx example
+<DialogTrigger type="popover" placement="bottom" shouldFlip>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover that will flip if it can't fully render below the button.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+### Container Padding
+
+You can control the minimum padding required between the Popover Dialog and the 
+surrounding container via the `containerPadding` prop. This affects the positioning
+breakpoints that determine when the Dialog will attempt to flip.
+
+The example below will flip the Dialog from above the trigger button to below the trigger
+button if the Dialog cannot render fully while maintaining 50px of padding between itself and
+the top of the browser.
+```tsx example
+<DialogTrigger type="popover" shouldFlip position="top" containerPadding={50}>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+## Events
+
+DialogTrigger accepts an `onOpenChange` handler which is triggered whenever the Dialog is opened or closed.
+
+The example below uses `onOpenChange` to update a separate `span` element with the current open state of the
+Dialog.
+
+```tsx example
+function Example() {
+  let [state, setState] = React.useState(false);
+
+  return (
+    <div>
+      <DialogTrigger type="popover" placement="top" onOpenChange={(isOpen) => setState(isOpen)}>
+        <ActionButton>Whispers</ActionButton>
+        <Dialog>
+          <Heading>Whispers and DMs</Heading>
+          <Divider />
+          <Content>
+            <Text>
+              You have 0 new messages.
+            </Text>
+          </Content>
+        </Dialog>
+      </DialogTrigger>
+      <span>Current open state: {state.toString()}</span>
+    </div>
+  );
+}
+```
+
+## Props
+
+<PropTable component={docs.exports.DialogTrigger} links={docs.links} />
+
+## Visual Options
+
+### Hide Arrow
+
+The `hideArrow` prop hides the Popover Dialog's arrow if applied.
+
+```tsx example
+<DialogTrigger hideArrow type="popover">
+  <ActionButton>See Notifications</ActionButton>
+  <Dialog>
+    <Heading>Notifications</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        You have no new notifications.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+### isOpen and defaultOpen
+
+The `isOpen` and `defaultOpen` props control whether the Dialog is open by default.
```

Comment:
why no examples? for menu trigger I found `isOpen` is a decent example, but `defaultOpen` won't stay open. :(

Reply:
I guess I can add an example of `isOpen`, just didn't feel right to have multiple dialogs open at the same time

Reply:
ah nevermind, the dialog always renders covering the example so I'll leave the example out

Reply:
That's a good point, I should probably remove the example from menu.